tests/test_infer/test_kvcache_manager.py

```diff
@@ -0,0 +1,165 @@
+import random
+from dataclasses import dataclass
+
+import torch
+from transformers.models.llama import LlamaConfig
+
+from colossalai.inference.kv_cache import CacheBlock, KVCacheManager
+from colossalai.logging import disable_existing_loggers
+from colossalai.testing import parameterize
+
+
+@dataclass
+class SampleInferenceConfig:
+    # This struct is only used for testing KVCache Manager.
+    # The inference config must have the following attributes to initialize a KVCacheManager.
+    block_size: int
+    max_batch_size: int
+    max_input_len: int
+    max_output_len: int
+    beam_width: int
+    dtype: torch.dtype
+    tp_size: int
+
+
+@parameterize(
+    "test_config",
+    [
+        {
+            "elem_size": 2,
+            "block_size": 4,
+        }
+    ],
+)
+def test_logical_blocks(test_config):
+    block = CacheBlock(block_id=0, block_size=test_config["block_size"], elem_size=test_config["elem_size"])
+
+    assert block.is_empty()
+    assert block.available_space == test_config["block_size"]
+    assert not block.has_ref()
+    block.add_ref()
+    assert block.ref_count == 1
+    assert block.has_ref()
+    block.remove_ref()
+    assert block.ref_count == 0
+    block.allocate(1)
+    assert block.allocated_size == 1
+    block.allocate(test_config["block_size"] - 1)
+    assert block.available_space < 1
+
+
+@parameterize(
+    "test_config",
+    [
+        {
+            "hidden_size": 512,
+            "num_attention_heads": 16,
+            "num_layers": 2,
+            "block_size": 8,
+            "max_batch_size": 10,
+            "max_input_len": 32,
+            "max_output_len": 32,
+            "dtype": torch.float32,
+            "beam_width": 1,
+            "tp_size": 1,
+        },
+        {
+            "hidden_size": 128,
+            "num_attention_heads": 4,
+            "num_layers": 3,
+            "block_size": 4,
+            "max_batch_size": 4,
+            "max_input_len": 64,
+            "max_output_len": 32,
+            "dtype": torch.float16,
+            "beam_width": 3,
+            "tp_size": 1,
+        },
+    ],
+)
+def test_cache_manager(test_config):
+    disable_existing_loggers()
+
+    assert test_config["max_batch_size"] > 1
+
+    hidden_size = test_config.pop("hidden_size")
+    num_layers = test_config.pop("num_layers")
+    num_attention_heads = test_config.pop("num_attention_heads")
+    head_size = hidden_size // num_attention_heads
+    block_size = test_config["block_size"]
+    max_batch_size = test_config["max_batch_size"]
+    max_input_length = test_config["max_input_len"]
+    max_output_length = test_config["max_output_len"]
+
+    sample_config = SampleInferenceConfig(**test_config)
+    model_config = LlamaConfig(
+        hidden_size=hidden_size,
+        num_hidden_layers=num_layers,
+        num_attention_heads=num_attention_heads,
+    )
+    cache_manager = KVCacheManager(sample_config, model_config)
```

Comment:
The one arg of `KVCacheManager` is an instance of `InferenceConfig`, why create the `SampleInferenceConfig` again?

Reply:
Done